tests/isbnbot/test_normalize_isbns.py

```diff
@@ -1,16 +1,138 @@
+            "9788496785441",
+            "9786084501138",
+            "9786086504540",
+        ],
+        "978-2-296-09310-2  978-84-96785-10-6. 978-84-96785-44-19789585596689": [
+            "9782296093102",
+            "9788496785106",
+            "9788496785441",
+            "9789585596689",
+        ],
+        "0-9752298-0-X   0 - 934314 - 08 - x ISBN   88-8191-235X": [
+            "097522980X",
+            "093431408X",
+            "888191235X",
+        ],
+    }
+    tricky = {
+        # same number twice, note dedupe is done after
+        "97815420429639781542042963": ["9781542042963", "9781542042963"],
+        # tab character
+        "\t978-83-8169-344-8": ["9788381693448"],
+        # unicode
+        "978\xad-85-\xad913035-\xad0-\xad2": ["9788591303502"],
+        "\u202c978-3-86850-235-0": ["9783868502350"],
+        "0-19-852X663-6": ["0198526636"],
+    }
+    invalid = [
+        "11111111112222222222222" "9881567445145",
+        "9771654981684",
+        "011249845X",
+        "7521245214",
+    ]
+    ignored = [
+        "978",
+        "-3-",
+        "-16-",
+        "-14810-0",
+        "$10.00",
+        "N/A",
+        "bought 2001-12-25",
+        "",
+        "13-978-951-558-248-5",
+        "13:978-1478322221",
+        "(ISBN13: 9781492993780",
+        "1567445144 (ISBN13: 9781567445145)",
+        "ISBN # 978-1-4243-0080-8 Registration Number TXu -1 - 333 – 097 Effective Date of Registration 11 Dec 2006",
+    ]
 
-    for isbn in ("ISBN 13: 9780425016015", "(ebook)0441788386"):
-        assert not NormalizeISBNJob.isbn_needs_normalization(isbn)
+    def test_single_inputs(self):
+        for input_str, expectedoutput in iter(self.single.items()):
+            with self.subTest("single input:" + input_str):
+                self.assertEqual(normalize_isbns.parse_isbns(input_str), expectedoutput)
+
+    def test_double_inputs(self):
+        for input_str, expectedoutput in iter(self.double.items()):
+            with self.subTest("double input:" + input_str):
+                self.assertEqual(normalize_isbns.parse_isbns(input_str), expectedoutput)
+
+    def test_multi_inputs(self):
+        for input_str, expectedoutput in iter(self.multi.items()):
+            with self.subTest("multi input:" + input_str):
+                self.assertEqual(normalize_isbns.parse_isbns(input_str), expectedoutput)
+
+    def test_tricky_inputs(self):
+        for input_str, expectedoutput in iter(self.tricky.items()):
+            with self.subTest("tricky input:" + input_str):
+                self.assertEqual(normalize_isbns.parse_isbns(input_str), expectedoutput)
+
+    def test_invalid_inputs(self):
+        for input_str in self.invalid:
+            with self.subTest("Invalid input:" + input_str):
+                self.assertFalse(normalize_isbns.parse_isbns(input_str))
+
+    def test_ignored_inputs(self):
+        for input_str in self.ignored:
+            with self.subTest("Should ignore:" + input_str):
```

Comment:
Python f-strings are faster, shorter, and easier to read.  https://www.scivision.dev/python-f-string-speed
```suggestion
            with self.subTest(f"single input:{input_str}"):
                self.assertEqual(normalize_isbns.parse_isbns(input_str), expectedoutput)

    def test_double_inputs(self):
        for input_str, expectedoutput in iter(self.double.items()):
            with self.subTest(f"double input:{input_str}"):
                self.assertEqual(normalize_isbns.parse_isbns(input_str), expectedoutput)

    def test_multi_inputs(self):
        for input_str, expectedoutput in iter(self.multi.items()):
            with self.subTest(f"multi input:{input_str}"):
                self.assertEqual(normalize_isbns.parse_isbns(input_str), expectedoutput)

    def test_tricky_inputs(self):
        for input_str, expectedoutput in iter(self.tricky.items()):
            with self.subTest(f"tricky input:{input_str}"):
                self.assertEqual(normalize_isbns.parse_isbns(input_str), expectedoutput)

    def test_invalid_inputs(self):
        for input_str in self.invalid:
            with self.subTest(f"Invalid input:{input_str}"):
                self.assertFalse(normalize_isbns.parse_isbns(input_str))

    def test_ignored_inputs(self):
        for input_str in self.ignored:
            with self.subTest(f"Should ignore:{input_str}"):
```